CreateService/handler.ts

```diff
@@ -42,32 +49,61 @@ import {
   retrievedServiceToApiService
 } from "../utils/conversions";
 import { ServicePayloadMiddleware } from "../utils/middlewares/service";
+import { UpsertServiceEvent } from "../utils/UpsertServiceEvent";
 
 type ICreateServiceHandler = (
   context: Context,
   auth: IAzureApiAuthorization,
   clientIp: ClientIp,
   userAttributes: IAzureUserAttributes,
   servicePayload: ApiService
-) => Promise<IResponseSuccessJson<ApiService> | IResponseErrorQuery>;
+) => Promise<
+  | IResponseSuccessJson<ApiService>
+  | IResponseErrorQuery
+  | IResponseErrorValidation
+>;
 
 export function CreateServiceHandler(
   _GCTC: CustomTelemetryClientFactory,
   serviceModel: ServiceModel
 ): ICreateServiceHandler {
-  return async (_, __, ___, ____, servicePayload) => {
+  return async (context, __, ___, ____, servicePayload) => {
     const service = apiServiceToService(servicePayload);
     const errorOrCreatedService = await serviceModel.create(
       service,
       service.serviceId
     );
-    return errorOrCreatedService.fold<
-      IResponseErrorQuery | IResponseSuccessJson<ApiService>
-    >(
-      error => ResponseErrorQuery("CreateServiceHandler error", error),
-      createdService =>
-        ResponseSuccessJson(retrievedServiceToApiService(createdService))
+
+    if (isLeft(errorOrCreatedService)) {
+      return ResponseErrorQuery(
+        "CreateServiceHandler error",
+        errorOrCreatedService.value
+      );
+    }
+
+    const createdService = errorOrCreatedService.value;
+
+    const errorOrUpsertServiceEvent = UpsertServiceEvent.decode({
```

Comment:
we should actually `encode` the typed object to an untyped one, this should be something like:

```ts
const upsertServiceOrchestratorInput = UpsertServiceEvent.encode({
  newService: createdService,
  updatedAt: new Date()
});

const dfClient = df.getClient(context);
    await dfClient.startNew(
      "UpsertServiceOrchestrator",
      undefined,
      upsertServiceOrchestratorInput
    );

```

Reply:
Done also for the Activity input.